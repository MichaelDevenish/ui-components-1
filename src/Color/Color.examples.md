### A Color Component
```jsx
const { brandColors } = require('./style');

<Color hex={brandColors.intelliRoyalBlue} description="intelli-royal-blue" />
```
