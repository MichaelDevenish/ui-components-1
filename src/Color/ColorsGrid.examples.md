### Brand Colours
```jsx
const { brandColorsArray } = require('./Color');

<ColorsGrid colors={brandColorsArray} />
```

### App Colours
```jsx
const { appColorsArray } = require('./Color');

<ColorsGrid colors={appColorsArray} />
```