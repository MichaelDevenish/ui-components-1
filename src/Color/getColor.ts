import colours from '../sass/resources/js/colours'

export const getColor = (name: string): string => {
  if (!colours[name]) {
    throw new Error(`Unknown Colour Name '${name}'`)
  }

  return colours[name]
}
