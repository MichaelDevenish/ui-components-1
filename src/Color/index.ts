export { Color } from './Color'
export { brandColors, appColors } from './style'
export {
  ColorsGrid
} from './ColorsGrid'
