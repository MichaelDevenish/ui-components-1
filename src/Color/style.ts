import React from 'react'
import styled, { StyledComponentClass } from 'styled-components'

export interface ColorWithNameAndValue {
  [key: string]: string
}

export interface ColorBoxProps {
  hex: string
}

export const brandColors: ColorWithNameAndValue = {
  intelliRoyalBlue: '#432DF3',
  intelliGreen: '#04DBAC',
  intelliYellow: '#FFDD37',
  intelliRed: '#FF4D50',
  intelliCyan: '#03DEEB',
  intelliPurple: '#A35AFF',
  intelliCharcoal: '#303035',
  intelliGrey1: '#B3B3B5',
  intelliGrey2: '#E4E4E5',
  intelliBaseWhite: '#F5F7F9',
  intelliOffWhite: '#F5F5F5'
}

export const appColors: ColorWithNameAndValue = {
  intelliAppBlue: '#0F68FA',
  intelliDarkGrey: '#47525D',
  intelliGreyBlue1: '#C3CCD0',
  intelliGreyBlue2: '#DFE5E8',
  intelliWhite: '#FFFFFF',
  intelliRoyalBlueLight: '#EBE9FE',
  intelliAppBlueLight: '#D7F2FF',
  intelliGreenLight: '#E0FEF8',
  intelliYellowLight: '#FFF9DA',
  intelliRedLight: '#FFE6E6'
}


export const ColorBox = styled.div`
  width: 10em;
  height: 7.5em;
  background-color: ${(props: ColorBoxProps) => props.hex};
  margin-bottom: 1em;
`

export const ColorName = styled.p`
  font-weight: 600;
`
export const WrappedList = styled.div`
  display: flex;
  flex-wrap: wrap;
  max-width: 50rem;
  margin-left: 0;
`
