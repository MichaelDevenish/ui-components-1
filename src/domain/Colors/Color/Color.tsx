import React from 'react'
import { map } from 'lodash'
import {
  WrappedList,
  ColorBox,
  ColorName
} from './style'
import { Variables } from '../../../common'

export interface IColorGridProps {
  colors: Variables.Color[]
}

export class ColorsGrid extends React.PureComponent<IColorGridProps> {
  public render (): JSX.Element {
    const { colors } = this.props

    return (
      <WrappedList>
        {map(colors, (color: Variables.Color, idx: number) =>
          (
            <div key={idx}>
              <ColorBox hex={color} />
              <ColorName> {color} </ColorName>
              <p> {color} </p>
            </div>
          )
        )}
      </WrappedList>
    )
  }
}
