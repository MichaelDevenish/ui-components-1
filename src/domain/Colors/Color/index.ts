export { Color, ColorObject, brandColorsArray } from './Color'
