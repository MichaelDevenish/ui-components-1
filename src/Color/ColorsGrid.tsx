import React from 'react'
import styled from 'styled-components'
import { ColorObject, Color } from './Color'
import { WrappedList } from './style'

export interface ColorGridProps {
  colors: ColorObject[]
}

export class ColorsGrid extends React.PureComponent<ColorGridProps> {
  render () {
    const { colors } = this.props

    return (
      <WrappedList>
        {colors.map(c => <Color hex={c.hex} description={c.name} key={c.name} />)}
      </WrappedList>
    )
  }
}
