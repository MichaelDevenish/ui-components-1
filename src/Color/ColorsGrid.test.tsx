import { mount } from 'enzyme'
import React from 'react'
import { ColorsGrid } from './'
import { brandColorsArray } from './Color'
import 'jest-styled-components'

describe('<ColorsGrid />', () => {
  it(`should render a color component`, () => {
    const wrapper = mount(
      <ColorsGrid colors={brandColorsArray} />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
