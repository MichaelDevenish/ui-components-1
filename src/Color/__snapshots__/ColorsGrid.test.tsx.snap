// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ColorsGrid /> should render a color component 1`] = `
.c1 {
  width: 10em;
  height: 7.5em;
  background-color: #432DF3;
  margin-bottom: 1em;
}

.c3 {
  width: 10em;
  height: 7.5em;
  background-color: #04DBAC;
  margin-bottom: 1em;
}

.c4 {
  width: 10em;
  height: 7.5em;
  background-color: #FFDD37;
  margin-bottom: 1em;
}

.c5 {
  width: 10em;
  height: 7.5em;
  background-color: #FF4D50;
  margin-bottom: 1em;
}

.c6 {
  width: 10em;
  height: 7.5em;
  background-color: #03DEEB;
  margin-bottom: 1em;
}

.c7 {
  width: 10em;
  height: 7.5em;
  background-color: #A35AFF;
  margin-bottom: 1em;
}

.c8 {
  width: 10em;
  height: 7.5em;
  background-color: #303035;
  margin-bottom: 1em;
}

.c9 {
  width: 10em;
  height: 7.5em;
  background-color: #B3B3B5;
  margin-bottom: 1em;
}

.c10 {
  width: 10em;
  height: 7.5em;
  background-color: #E4E4E5;
  margin-bottom: 1em;
}

.c11 {
  width: 10em;
  height: 7.5em;
  background-color: #F5F7F9;
  margin-bottom: 1em;
}

.c12 {
  width: 10em;
  height: 7.5em;
  background-color: #F5F5F5;
  margin-bottom: 1em;
}

.c2 {
  font-weight: 600;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  max-width: 50rem;
  margin-left: 0;
}

<ColorsGrid
  colors={
    Array [
      Object {
        "hex": "#432DF3",
        "name": "intelli-royal-blue",
      },
      Object {
        "hex": "#04DBAC",
        "name": "intelli-green",
      },
      Object {
        "hex": "#FFDD37",
        "name": "intelli-yellow",
      },
      Object {
        "hex": "#FF4D50",
        "name": "intelli-red",
      },
      Object {
        "hex": "#03DEEB",
        "name": "intelli-cyan",
      },
      Object {
        "hex": "#A35AFF",
        "name": "intelli-purple",
      },
      Object {
        "hex": "#303035",
        "name": "intelli-charcoal",
      },
      Object {
        "hex": "#B3B3B5",
        "name": "intelli-grey-1",
      },
      Object {
        "hex": "#E4E4E5",
        "name": "intelli-grey-2",
      },
      Object {
        "hex": "#F5F7F9",
        "name": "intelli-base-white",
      },
      Object {
        "hex": "#F5F5F5",
        "name": "intelli-off-white",
      },
    ]
  }
>
  <styled.div>
    <div
      className="c0"
    >
      <Color
        description="intelli-royal-blue"
        hex="#432DF3"
        key="intelli-royal-blue"
      >
        <div>
          <styled.div
            hex="#432DF3"
          >
            <div
              className="c1"
            />
          </styled.div>
          <styled.p>
            <p
              className="c2"
            >
               
              intelli-royal-blue
               
            </p>
          </styled.p>
          <p>
             
            #432DF3
             
          </p>
        </div>
      </Color>
      <Color
        description="intelli-green"
        hex="#04DBAC"
        key="intelli-green"
      >
        <div>
          <styled.div
            hex="#04DBAC"
          >
            <div
              className="c3"
            />
          </styled.div>
          <styled.p>
            <p
              className="c2"
            >
               
              intelli-green
               
            </p>
          </styled.p>
          <p>
             
            #04DBAC
             
          </p>
        </div>
      </Color>
      <Color
        description="intelli-yellow"
        hex="#FFDD37"
        key="intelli-yellow"
      >
        <div>
          <styled.div
            hex="#FFDD37"
          >
            <div
              className="c4"
            />
          </styled.div>
          <styled.p>
            <p
              className="c2"
            >
               
              intelli-yellow
               
            </p>
          </styled.p>
          <p>
             
            #FFDD37
             
          </p>
        </div>
      </Color>
      <Color
        description="intelli-red"
        hex="#FF4D50"
        key="intelli-red"
      >
        <div>
          <styled.div
            hex="#FF4D50"
          >
            <div
              className="c5"
            />
          </styled.div>
          <styled.p>
            <p
              className="c2"
            >
               
              intelli-red
               
            </p>
          </styled.p>
          <p>
             
            #FF4D50
             
          </p>
        </div>
      </Color>
      <Color
        description="intelli-cyan"
        hex="#03DEEB"
        key="intelli-cyan"
      >
        <div>
          <styled.div
            hex="#03DEEB"
          >
            <div
              className="c6"
            />
          </styled.div>
          <styled.p>
            <p
              className="c2"
            >
               
              intelli-cyan
               
            </p>
          </styled.p>
          <p>
             
            #03DEEB
             
          </p>
        </div>
      </Color>
      <Color
        description="intelli-purple"
        hex="#A35AFF"
        key="intelli-purple"
      >
        <div>
          <styled.div
            hex="#A35AFF"
          >
            <div
              className="c7"
            />
          </styled.div>
          <styled.p>
            <p
              className="c2"
            >
               
              intelli-purple
               
            </p>
          </styled.p>
          <p>
             
            #A35AFF
             
          </p>
        </div>
      </Color>
      <Color
        description="intelli-charcoal"
        hex="#303035"
        key="intelli-charcoal"
      >
        <div>
          <styled.div
            hex="#303035"
          >
            <div
              className="c8"
            />
          </styled.div>
          <styled.p>
            <p
              className="c2"
            >
               
              intelli-charcoal
               
            </p>
          </styled.p>
          <p>
             
            #303035
             
          </p>
        </div>
      </Color>
      <Color
        description="intelli-grey-1"
        hex="#B3B3B5"
        key="intelli-grey-1"
      >
        <div>
          <styled.div
            hex="#B3B3B5"
          >
            <div
              className="c9"
            />
          </styled.div>
          <styled.p>
            <p
              className="c2"
            >
               
              intelli-grey-1
               
            </p>
          </styled.p>
          <p>
             
            #B3B3B5
             
          </p>
        </div>
      </Color>
      <Color
        description="intelli-grey-2"
        hex="#E4E4E5"
        key="intelli-grey-2"
      >
        <div>
          <styled.div
            hex="#E4E4E5"
          >
            <div
              className="c10"
            />
          </styled.div>
          <styled.p>
            <p
              className="c2"
            >
               
              intelli-grey-2
               
            </p>
          </styled.p>
          <p>
             
            #E4E4E5
             
          </p>
        </div>
      </Color>
      <Color
        description="intelli-base-white"
        hex="#F5F7F9"
        key="intelli-base-white"
      >
        <div>
          <styled.div
            hex="#F5F7F9"
          >
            <div
              className="c11"
            />
          </styled.div>
          <styled.p>
            <p
              className="c2"
            >
               
              intelli-base-white
               
            </p>
          </styled.p>
          <p>
             
            #F5F7F9
             
          </p>
        </div>
      </Color>
      <Color
        description="intelli-off-white"
        hex="#F5F5F5"
        key="intelli-off-white"
      >
        <div>
          <styled.div
            hex="#F5F5F5"
          >
            <div
              className="c12"
            />
          </styled.div>
          <styled.p>
            <p
              className="c2"
            >
               
              intelli-off-white
               
            </p>
          </styled.p>
          <p>
             
            #F5F5F5
             
          </p>
        </div>
      </Color>
    </div>
  </styled.div>
</ColorsGrid>
`;
